src/analyzer/analyzerwaveform.cpp

```diff
@@ -23,7 +25,9 @@ AnalyzerWaveform::AnalyzerWaveform(UserSettingsPointer pConfig) :
     m_filter[1] = 0;
     m_filter[2] = 0;
 
+    QMutexLocker lock(&s_mutex);
```

Comment:
Why don't you use a QAtomicInt for this purpose? Mixing a static class member with static local function variable is uncommon.

The QAtomicInt can be defined in an anonymous namespace inside this .cpp file. There is no need to expose it as a static class member.